FreeRTOS/FreeRTOS-Plus-TCP
# IPv6/Multi: Reduced complexity FreeRTOS_DHCP

Description
-----------
The IPv4 version of DHCP also had a few fairly complex functions. 

 vDHCPProcessEndPoint()  score 104
 - xHandleWaitingOffer()
 - vHandleWaitingAcknowledge()
 - xHandleWaitingFirstDiscover()
 - prvHandleWaitingeLeasedAddress()
 
 prvProcessDHCPReplies()  score **269**
 - vProcessHandleOption()
 - xProcessCheckOption()

After the PR, the scores will be about 17. I think it doesn't help clarity if we break them down further into smaller parts.

Test Steps
-----------
I tested the changes by running DHCP.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
